### ISSUE TITLE
Point to profile data

### ISSUE THREAD

**Issue by dhruvbalwada** [GliderToolsCommunity/GliderTools#30]
**Is your feature request related to a problem? Please describe.**
It would be great to able to convert the data from point to profile type. Currently, the data is saved as a long series of variables. It is quite natural instead to sometimes refer to data in terms of dive numbers. 
This can also help resolve the issue that we currently always look at the down casts in order of increasing Z and upcasts in decreasing Z, which can create visual artifacts in scatter plots. 

**Describe the solution you'd like**
The simplest solution would be to adopt the points-profiles style that [argopy](argopy.readthedocs.io) is using in https://argopy.readthedocs.io/en/latest/_modules/argopy/xarray.html#ArgoAccessor.point2profile

Maybe we don't need to have data types, but a simpler (maybe more hacky) solution would be great for an initial solution.

**Comment by luke-gregor** [comment]
Potential hack solution to create a dictionary using groupby on the dives. Might be slow though. Also, does not provide one with any functions that can be applied to each dive

```python
import glidertools as gt

xds = gt.load.seaglider_basestation_netCDFs('Git/GliderTools/tests/data/p54203*.nc', ['temperature'])['sg_data_point']

grp = xds.groupby('dives')

dim = list(xds.dims.keys())[0]
out = {}
for k, i in grp.groups.items():
    out[k] = xds.isel(**{dim: i}).sortby('ctd_depth')
```

**Comment by MartinMohrmann** [comment]
Hi, I am aware that this is an old issue - but I fought with it to an came up with some good solutions :). An easy and fast way to convert the point-data to profile-data is the groupby function as @lukegre mentioned. Once the data is grouped into profiles, any function can be applied with the pandas/xarray `map` or `apply` methods. If used on dask-arrays, the per-profile functions run in parallel which makes processing very fast.

 ```
ds.groupby("dives"
    ).apply(function, input_variables)
```
I also have a considerable speed increase if my ds-object is a `pandas.Dataframe` instead of a `xarray.Dataset`. Casting can be done by `ds.to_pandas()` or `ds.reset_coords().to_pandas()` if your dataset had multiple coordinates from the start.

**Comment by MartinMohrmann** [comment]
This was addressed in PR #175, have a look at the new version. 
Closed, feel free to reopen.
